src/re_frame/interop.clj

```diff
@@ -0,0 +1,43 @@
+(ns re-frame.interop)
+
+(defn next-tick [f]
+  (.start (Thread. f)))
+
+(def empty-queue clojure.lang.PersistentQueue/EMPTY)
+
+(def after-render next-tick)
+
+(def debug-enabled? true)
+
+(defn ratom [x]
+  (let [result (atom x)]
+    (alter-meta! result assoc ::ratom? true)
```

Comment:
I'm unsure what the best way to approach this is.  Creating a new protocol and extending it to `IAtom` would mean that we don't detect when you've just passed in a raw `clojure.core/atom` -- in which case there's no point in a protocol, we might as well just test whether you've passed in an atom.

What I've done here seems like the easiest way to detect whether you've used `re-frame.interop/ratom` to create the atom.  But I'm not sure how useful that check actually is in the first place.  The `ratom?` function gets used to sanity-check that `pure`, `subscribe` and `fx` get given `app-db`, not some other random thing.  I vaguely wonder whether in testing, it's more appropriate just to say "yup, I got some sort of atom, that's good enough" and leave the Reactive-ness part of the check to the CLJS runtime?

So I'm inclined to replace the CLJ implementation of `ratom` with just `(atom x)`, getting rid of the whole metadata thing.